### DISCUSSION TITLE
Let `quarto render` not clear target folder entirely

### DISCUSSION THREAD

**Discussion by psychelzh** [quarto-dev/quarto-cli#5482] [Q&A] [answered]
I am treating the rendered website folder as a submodule, so every time when I render the site, I need to initiate that submodule again, because the render process will remove the submodule entirely. Is there a way to keep those files specific to git?

**Comment by mcanouil** [comment, marked as answer]
You need to specify website resources, see <https://quarto.org/docs/websites/website-tools.html#site-resources>.

**Reply by psychelzh** [reply]
Thank you very much. The first thing I tried is to add `.git` to resources. But unfortunately, there is some permission issue here:

```bash
Rendering:
[1/3] index.qmd
[2/3] posts\compare_modality.qmd
[3/3] posts\main_results.qmd

ERROR: PermissionDenied: 拒绝访问。 (os error 5), remove 'D:\OneDrive\Documents\Research\brain-intelligence\g-factor\docs\.git\modules\docs\objects\04\2e2dc36771b29fae14b72e5798fc3e69138228'

PermissionDenied: 拒绝访问。 (os error 5), remove 'D:\OneDrive\Documents\Research\brain-intelligence\g-factor\docs\.git\modules\docs\objects\04\2e2dc36771b29fae14b72e5798fc3e69138228'
    at Object.removeSync (deno:runtime/js/30_fs.js:157:9)
    at copyFileSync (file:///C:/PROGRA~1/Quarto/bin/quarto.js:58534:14)
    at copyTo (file:///C:/PROGRA~1/Quarto/bin/quarto.js:58481:9)
    at doCopy (file:///C:/PROGRA~1/Quarto/bin/quarto.js:58511:9)
    at copyFileIfNewer (file:///C:/PROGRA~1/Quarto/bin/quarto.js:58528:9)
    at copyResourceFile (file:///C:/PROGRA~1/Quarto/bin/quarto.js:59238:5)
    at file:///C:/PROGRA~1/Quarto/bin/quarto.js:86961:25
    at Set.forEach (<anonymous>)
    at file:///C:/PROGRA~1/Quarto/bin/quarto.js:86958:30
    at Array.forEach (<anonymous>)
```

**Reply by mcanouil** [reply]
It's a well known issue with cloud drive and external drive.

In addition, it's not a great idea to have version control within a tool that has version control.

**Reply by psychelzh** [reply]
Thanks. Yes, it is probably because `.git` is also in the main directory, and `quarto` tries to copy that to the submodule. Maybe just as what you said, treating the generated folder as a submodule is not a great idea.

**Reply by mcanouil** [reply]
My point was mostly on OneDrive + Git.
OneDrive (and most of cloud storage services) does version control, not the same way, but still.

But, Git submodule even if they seem nice, they have a lot of caveats and in my opinion should be used after really thinking about it.

Overall, moving your project out or OneDrive, adding your external resources to `resources` yaml key in `_quarto.yml` or in the header of relevant document should solve your issue.
